path-finding.asciidoc

```diff
@@ -12,3 +12,31 @@ Relevant questions to answer:
   * What information can be sent to intermediate and the final node aside from the critical routing data?
   * What are multi-hop locks? What addition privacy and security guarantees to they offer?
   * How can the flexible onion space be used to enabled packet switching in the network?
+
+
+=== What is "Source-Based" routing and why does the Lightning Network use it?
+
+Source-based routing is a method of path-finding where the sender (i.e. the source) plans the route from itself, through the intermediary nodes, and finally to the destination.
+Once a route has been selected, the sender sends the payment to the first intermediary node, who sends it to the second intermediary node and so on, until it reaches the destination.
+While a payment is in-flight along a route, the route cannot be changed by any of the intermediary nodes, even if a shorter path or a cheaper path (in terms of routing fees) exists.
+
+The Lightning Network uses source-based routing at the time of writing in order to protect user privacy.
+As discussed in the chapter on Onion Routing, the intermediary nodes transmitting the payment are not aware of the full route of the payment; they only know the node they received it from and the node they are sending it to.
+As such, intermediary nodes cannot assist with path-finding because this requires them to know the destination of the payment.
+Without knowing the destination, they cannot calculate whether the path selected is efficient or not.
+
+We also cannot expect the destination to find a path.
+Like Bitcoin, the Lightning Network uses a "push" system to make payments.
+In Bitcoin, payments are pushed by the sender to the receiver's address, and in Lightning, payments are pushed from the sender's channel to fulfill the receiver's invoice.
+The recipient is unable to "pull" a payment from anyone.
+Even if it specifies a route in the invoice, that route may no longer be viable by the time the invoice is paid, which could be several minutes or several days later.
+The recipient can, however, specify "routing hints" in the invoice that may assist the sender in finding a possible path.
+
+Furthermore, source-based routing comes with some inherent drawbacks.
+The sender chooses the route based on their current understanding of the topological map of the Lightning network.
+As discussed in previous chapters, this map is necessarily incomplete; the sender may not be aware of all the channels, and even if they are they will almost certainly not know the latest balances in each of the channels.
+And even if the sender did have a complete topological map at one point in time, the balances of channels change with every payment, and so in a short space of time the map would become obsolete.
+It is for this reason that the sender probes various paths until it finds a satisfactory path, and then sends the payment.
```

Comment:
probing and sending is kind of equivelent. the user could concurrently probe various paths with various fake payment hashes and then send along a path that seemed to have enough liquidity

Reply:
Hmm explain? I'm not too familiar with the mechanics of probing. I'm currently under the impression that you basically ask a bunch of nodes about their capacity and then try to figure out if the payment will work. Not sure how the fake payment hashes work. Could you suggest a more precise wording?

https://github.com/lnbook/lnbook/pull/309/commits/eed081a19fc3746ec0168bb207aea7e591869ebe